src/parallel.cpp

```diff
@@ -435,10 +441,12 @@ void CParallelValidation::ClearOrphanCache(const CBlock &block)
 //  has finished.
 void CParallelValidation::HandleBlockMessage(CNode *pfrom,
     const string &strCommand,
-    const CBlock &block,
-    const CInv &inv)
+    std::shared_ptr<CBlock> block,
+    const CInv &inv,
+    uint64_t nBlockSize)
 {
-    uint64_t nBlockSize = ::GetSerializeSize(block, SER_NETWORK, PROTOCOL_VERSION);
+    if (nBlockSize == 0)
+        nBlockSize = ::GetSerializeSize(*block, SER_NETWORK, PROTOCOL_VERSION);
```

Comment:
why not makes this PR depends on #959 and use `GetBlockSize()` rather than `::GetSerializeSize`?